web/server/codechecker_server/api/report_server.py

```diff
@@ -2157,31 +2158,49 @@ def changeReviewStatus(self, report_id, status, message):
     @timeit
     def getReviewStatusRules(self, rule_filter, sort_mode, limit, offset):
         self.__require_view()
+        if not sort_mode:
+            sort_mode = ReviewStatusRuleSortMode(
+                type=ReviewStatusRuleSortType.DATE,
+                ord=Order.DESC)
 
         result = []
-        with DBSession(self._Session) as session:
-            if not sort_mode:
-                sort_mode = ReviewStatusRuleSortMode(
-                    type=ReviewStatusRuleSortType.DATE,
-                    ord=Order.DESC)
 
-            q = get_rs_rule_query(session, rule_filter, sort_mode)
+        # To avoid modifiying the collection due to chunking
+        rule_filter_copy = deepcopy(rule_filter)
+
+        def getRules(reportHashes=None):
+            if rule_filter and reportHashes:
+                rule_filter_copy.reportHashes = reportHashes
+            q = get_rs_rule_query(session, rule_filter_copy, sort_mode)
 
             if limit:
                 q = q.limit(limit).offset(offset)
 
-        for review_status, associated_report_count in q:
-            result.append(ReviewStatusRule(
-                reportHash=review_status.bug_hash,
-                reviewData=create_review_data(
-                    review_status.status,
-                    review_status.message,
-                    review_status.author,
-                    review_status.date,
-                    False),
-                associatedReportCount=associated_report_count))
+            for review_status, associated_report_count in q:
+                result.append(ReviewStatusRule(
+                    reportHash=review_status.bug_hash,
+                    reviewData=create_review_data(
+                        review_status.status,
+                        review_status.message,
+                        review_status.author,
+                        review_status.date,
+                        False),
+                    associatedReportCount=associated_report_count))
+            return result
 
-        return result
+        with DBSession(self._Session) as session:
+            if rule_filter and rule_filter.reportHashes:
+                # Diffing with a large ammount of report hashes passed in the
+                # filter (60K) caused a hanging report diffing.
+                # The probable cause of this is the ILIKE matching in the
+                # underlying logic. Chunking the request solves this issue.
+                for hash_chunk in util.chunks(
+                        iter(rule_filter.reportHashes),
```

Comment:
No need to use `iter()` here, because it is applied in `util.chunk()`. I think we shouldn't expect the caller to know what kind of iterable can be given to `util.chunk()`.

Reply:
True, this modification was around the time we did the chunks fix. Will remove.